File: app/graphql/functions/comment_photo.rb
class Functions::CommentPhoto < GraphQL::Function
  argument :content, !types.String
  argument :photo_id, !types.ID

  type Types::CommentType

  def call(obj, args, ctx)
    if user = ctx[:current_user]
      photo = Photo.find(args[:photo_id])

      photo.comments.create!(content: args[:content], user: user)
    else
      GraphQL::ExecutionError.new("Unauthorized")
    end
  end
end

File: app/javascript/packs/mutations/comment_photo.js
import { gql } from "react-apollo"

export default gql`
  mutation commentPhoto($photo_id: ID!, $content: String!) {
    commentPhoto(photo_id: $photo_id, content: $content) {
      id
      content
      user {
        id
        username
      }
    }
  }
`
